src/main/java/org/candlepin/subscriptions/resource/SubscriptionTableController.java

```diff
@@ -377,39 +379,29 @@ public static void addTotalCapacity(
 
     var metric = key.getMetricId();
     var type = key.getMeasurementType();
-
-    var sockets =
-        (SOCKETS.equalsIgnoreCase(metric) && "PHYSICAL".equals(type)) ? value.intValue() : 0;
-    var cores = (CORES.equalsIgnoreCase(metric) && "PHYSICAL".equals(type)) ? value.intValue() : 0;
-
-    var hypervisorSockets =
-        (SOCKETS.equalsIgnoreCase(metric) && "HYPERVISOR".equals(type)) ? value.intValue() : 0;
-    var hypervisorCores =
-        (CORES.equalsIgnoreCase(metric) && "HYPERVISOR".equals(type)) ? value.intValue() : 0;
-
-    if (skuCapacity.getUom() == Uom.SOCKETS || SOCKETS.equals(skuCapacity.getMetricId())) {
-      skuCapacity.setCapacity(skuCapacity.getCapacity() + sockets);
-      skuCapacity.setHypervisorCapacity(skuCapacity.getHypervisorCapacity() + hypervisorSockets);
-    } else if (skuCapacity.getUom() == Uom.CORES || CORES.equals(skuCapacity.getMetricId())) {
-      skuCapacity.setCapacity(skuCapacity.getCapacity() + cores);
-      skuCapacity.setHypervisorCapacity(skuCapacity.getHypervisorCapacity() + hypervisorCores);
-    } else if (sockets != 0 || hypervisorSockets != 0) {
-      skuCapacity.setCapacity(skuCapacity.getCapacity() + sockets);
-      skuCapacity.setHypervisorCapacity(skuCapacity.getHypervisorCapacity() + hypervisorSockets);
-      if (skuCapacity.getUom() == null) {
-        skuCapacity.setUom(Uom.SOCKETS);
-      }
+    // we only initialize the metric for measurements with a value higher than zero.
+    if (value > 0) {
+      // initialize metric ID using the current metric if it's not set
       if (skuCapacity.getMetricId() == null) {
-        skuCapacity.setMetricId(SOCKETS);
+        skuCapacity.setMetricId(metric);
       }
-    } else if (cores != 0 || hypervisorCores != 0) {
-      skuCapacity.setCapacity(skuCapacity.getCapacity() + cores);
-      skuCapacity.setHypervisorCapacity(skuCapacity.getHypervisorCapacity() + hypervisorCores);
+      // initialize uom using the current metric if it's not set
       if (skuCapacity.getUom() == null) {
-        skuCapacity.setUom(Uom.CORES);
+        if (SOCKETS.equalsIgnoreCase(metric)) {
+          skuCapacity.setUom(Uom.SOCKETS);
+        } else if (CORES.equalsIgnoreCase(metric)) {
+          skuCapacity.setUom(Uom.CORES);
```

Comment:
How would you feel about guarding against nonsensical requests like:

```
http://localhost:8000/api/rhsm-subscriptions/v1/subscriptions/products/rosa?metric_id=Instance-hours&uom=sockets
```

I'm wondering if it's worth throwing a BadRequestException as a final else on here.

Reply:
I would not over complicate the logic here since the uom param is deprecated and will be deleted. Because the uom query param is deprecated, the metricId takes priority. 
Indeed, this scenario that you describe can happen nowadays (without these changes) and I dehistate to introduce potential breaking changes, but let me know if you strongly feel we should add it, and will do so.